### DISCUSSION TITLE
SumatraPDF returns exit code 1

### DISCUSSION THREAD

**Discussion by ryder-james** [sumatrapdfreader/sumatrapdf#3021] [Q&A]
Hi all. I'm running SumatraPDF using Go's os/exec package. When I run `SumatraPDF.exe -reuse-instance -page <page-no> <file>`, SumatraPDF launches to the correct page and works as expected, but returns exit code 1. The program I'm running throws an error if the command returns anything other than 0, so I'm trying to get to the bottom of why it's returning 1 to see if there's anything I can do about it.

**Comment by GitHubRulesOK** [comment]
several times users say Sumara exit state is not correct, however that is often down to shell handling, run the command 

`SumatraPDF.exe -log -reuse-instance -page <page-no> <file>` close and read last line

I see `Exiting with exit code: 0` even with bad switch or bad filename raising an internal error, the closure is clean
Perhaps check if similar last line

**Reply by ryder-james** [reply]
Unfortunately the log switch doesn't change anything, the return code is still 1 but there is no logging or errors of any kind.

**Reply by ryder-james** [reply]
![image](https://user-images.githubusercontent.com/5988209/195478267-8c4c094a-1620-40b3-b0e6-19636beb8d23.png)

**Reply by GitHubRulesOK** [reply]
you don't appear to be running under `standard windows command line console` (normal %ComSpec%)?
\>sumatrapdf -log will exit with exit code 0 showing by `notepad C:\%USERPROFILE%\AppData\Local\sumatra-log.txt`

hence I said it is likely the shell see https://github.com/msys2/MSYS2-packages/issues/2869

**Reply by ryder-james** [reply]
Done some more investigating. First off, I get the same issue when running through cmd rather than through git bash - though the error code is 9009. It will work as normal in appearance, but it returns a non-zero exit code, causing my other piece of software to recognize it as a failure.
Further, it appears that it only returns a non-zero exit code if used while SumatraPDF is already open - if SumatraPDF is not yet open, it will open to the correct page of the correct file with a 0 exit code.

I also do not have a C:/%USERPROFILE%/AppData/Local/sumatra-log.txt, even after running sumatrapdf -log (this is true of both cmd and of Git Bash/MINGW64)

**Reply by GitHubRulesOK** [reply]
>I also do not have a C:/%USERPROFILE%/AppData/Local/sumatra-log.txt, even after running sumatrapdf -log (this is true of both cmd and of Git Bash/MINGW64)

sorry forgot -log is  newer thus guessing your using a 3.4# version 

any dirrect way I use the console command (good and bad) Sumatrapdf returns zer0 on exit, here the fail is a bad cmd call
![image](https://user-images.githubusercontent.com/12754913/197070524-93f99c20-fb0b-4325-8fee-ff1f2f66aceb.png)


most calling methods should return 0 so variations on above fail
![image](https://user-images.githubusercontent.com/12754913/197071838-c76ad41c-8ca2-4840-80ac-3ebe26cc9554.png)

multiple calls including first bad then two good

![image](https://user-images.githubusercontent.com/12754913/197073462-ae14b502-07d2-4a2e-a636-c8d597ad7f59.png)


Multiple  direct two good and a bad one for internal error msg

![image](https://user-images.githubusercontent.com/12754913/197073916-5ea605f0-e3a3-49d7-a953-eacc4eb46cd9.png)

**Comment by ilyaza** [comment]
This was reported (and confirmed by the Spdf team) long time ago!  See
    https://forum.sumatrapdfreader.org/t/perl-errorlevels-with-256-when-opening-a-document-on-windows/4556

**Reply by ryder-james** [reply]
It's great to see it's been somewhat discussed, has any sort of workaround been found?
